### ISSUE TITLE
Support for `\vspace`

### ISSUE THREAD

**Issue by claell** [michal-h21/tex4ebook#125]
Recently, I discovered, that currently, there doesn't seem to be support for the `\vspace` command.

I assume this is on purpose, similar to #100.

However, especially with the option `p-indent`, that causes visual problems when one wants to visually separate some paragraphs from others (also see #122 for the similar issue that is about horizontal margins).

I hope, some good solution for this can be found.

**Comment by michal-h21** [comment]
I think it is better to use custom commands in this case. `\vspace` is used internally by many commands, so it could lead to unexpected results if you always insert vertical space for it. See [this answer](https://tex.stackexchange.com/a/314854/2891) for some examples.

**Comment by claell** [comment]
Yes, I understand and agree. Ideally, something like in #117 could be provided to ensure robustness of code, maintainability and ease of use.

**Comment by claell** [comment]
Based on the given material, I have (theoretically, untested) came up with something like:

```
\renewcommand\epubvspace{\a:epubvspace}
\NewConfigure{epubvspace}{1}
\Configure{epubvspace}{\ifvmode\HCode{<span class="vspace">}\:nbsp\HCode{</span>}\fi}
\Css{.vspace{height:#1;margin:0;}}
```

Is that sensible (does it work, and other improvement suggestions)?

**Comment by michal-h21** [comment]
For this LaTeX package, this `.4ht` file should work:

```
\def\pt:to:em#1{\strip@pt\dimexpr#1/\f@size em}
\renewcommand\epubvspace[1]{\edef\:vspacesize{\pt:to:em{#1}}\a:epubvspace}
\NewConfigure{epubvspace}{1}
\Configure{epubvspace}{\ifvmode\IgnorePar\fi\EndP\HCode{<div class="vspace" style="height:\:vspacesize">}\:nbsp\HCode{</div>}}
\Css{.vspace + p{margin-top:0pt;}}
\Css{p + .vspace{margin-top:0pt;}}
```

It saves the amount specified for the `\epubvspace` command into `\:vspacesize` command, so it can be reused in the configuration. It also converts any specified dimension to `em` units. `\Css` commands removes default vertical spaces between `div` and `p`, so only the specified amount is used.

**Comment by claell** [comment]
Nice, perfect. Many thanks!

Right now, I am not sure, if I am doing things right. I have included the command creation in a LaTeX style that gets loaded in the main LaTeX document.

And I also created `epubvspace.4ht` and placed it next to the main folder with the LaTeX file where `tex4ebook` gets called.

Unfortunately, it doesn't seem to have an effect. Do I need to place the `.4ht` file somewhere else?

**Comment by michal-h21** [comment]
Is the main style named `epubvspace.sty`? If not, you may need to rename the `epubvspace.4ht` to the same basename. If they are same, can you find in the `.log` file that `epubvspace.4ht` is loaded?

**Comment by claell** [comment]
Ah, okay, didn't know that. The main style is at a sub folder (possibly, that might also cause further conflicts) and is just named `book.sty`.

So I'll try with `book.4ht` for now.

**Comment by claell** [comment]
Interesting. Now, the spacing seems to work, at least. But it broke many other things, like the TOC, etc.

**Comment by claell** [comment]
I assume, `book.4ht` conflicts with another "package" that has the same name?

**Comment by michal-h21** [comment]
Well, it clashes with `book.cls`. I would suggest to rename the package to something different, for example `mybook.sty` and `mybook.4ht`. This should prevent errors.

**Comment by claell** [comment]
Mh, I renamed as you suggested. The weird thing is that now, the `\epubvspace` command doesn't get applied in the EPUB.

I'll try to check whether I can fix that by moving the location of `mybook.sty` (which was in a subfolder, previously) or moving `mybook.4ht` to that subfolder.

**Comment by michal-h21** [comment]
In this case, you should put the  `.4ht` file to the same directory as your `.sty` file.

**Comment by claell** [comment]
So for the flawed integration of that style:

Before, I did it like:

```latex
\usepackage{./subfolder/mybook}
```

and in the style:


```latex
\ProvidesPackage{mybook}
```

which always gave the warning

```
LaTeX: You have requested package `./subfolder/mybook',
               but the package provides `mybook'.
```

So I fixed that with changing the style to

```latex
\ProvidesPackage{./subfolder/mybook}
```

Unfortunately, that still doesn't make the `mybook.4ht` get used when it is next to `mybook.sty` inside of `subfolder`.

**Comment by claell** [comment]
Thanks, that works!

**Comment by claell** [comment]
After checking back with my PDF, I noticed, that sometimes, `vspaces` follow each other, in which case, I don't want them to add up. So actually, I need to use `\addvspace` for that (https://tex.stackexchange.com/questions/398590/ignore-vertical-spaces-that-follow-one-another).

Similarly, support for `\epubaddvspace` would be great!

I have been experimenting a bit with margins and other options, but didn't get a working result, yet. I think, collapsible margins might be a way to achieve it.

**Comment by michal-h21** [comment]
I think in HTML, this CSS definition should suppress additional spaces after multiple `\epubvspace` commands that immediately follows each other:

     \Css{.vspace + .vspace{height:0;margin:0;}}

**Comment by claell** [comment]
Thanks! As that topic seems to be also a bit more expanding, I created a follow-up issue: #130.
